src/pkg/bundle/remote.go

```diff
@@ -252,35 +258,29 @@ func (op *ociProvider) LoadBundle(_ int) (PathMap, error) {
 	}
 	layersToPull = append(layersToPull, rootDesc)
 
-	// only grab image layers that we want
-	// would like to use oci.CopyWithProgress here but it breaks when the media type of the image manifest is a Zarf blob
-	for _, layer := range layersToPull {
-		spinner.Updatef(fmt.Sprintf("Pulling bundle layer: %s", layer.Digest.Encoded()))
-		if ok, _ := op.Repo().Exists(op.ctx, layer); ok {
-			lb, err := op.Repo().Fetch(op.ctx, layer)
-			if err != nil {
-				return nil, err
-			}
-			exists, err := store.Exists(op.ctx, layer)
-			if err != nil {
-				return nil, err
-			}
-			if !exists {
-				err = store.Push(op.ctx, layer, lb)
-				if err != nil {
-					return nil, err
-				}
-			}
-		}
+	// copy bundle
+	copyOpts, estimatedBytes, err := utils.CreateCopyOpts(layersToPull, config.CommonOptions.OCIConcurrency)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a thread to update a progress bar as we save the package to disk
+	doneSaving := make(chan int)
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go zarfUtils.RenderProgressBarForLocalDirWrite(op.dst, estimatedBytes, &wg, doneSaving, fmt.Sprintf("Pulling bundle: %s", bundle.Metadata.Name))
+	_, err = oras.Copy(op.ctx, op.Repo(), op.Repo().Reference.String(), store, op.Repo().Reference.String(), copyOpts)
+	if err != nil {
+		return nil, err
```

Comment:
I am maybe a 2/10 in Golang these days, but will this leak goroutines when there are errors?

Reply:
Good call! I'll add the trigger to stop that goroutine if an error is detected